app/queue/notifications.py

```diff
@@ -212,14 +212,12 @@ def build_base_notification_obj(notification_type: str, host: dict):
     if notification_type == NotificationType.validation_error.value:
         return base_obj
 
-    canonical_facts = host.get("canonical_facts", deserialize_canonical_facts(host))
-
     system_profile = host.get("system_profile_facts", host.get("system_profile", {}))
 
     complete_base_obj: dict[str, Any] = {
         "context": {
             "inventory_id": host.get("id"),
-            "hostname": canonical_facts.get("fqdn", ""),
+            "hostname": host.get("fqdn") or "",
```

Comment:
I'd prefer to keep the `get` semantic.

```suggestion
            "hostname": host.get("fqdn", ""),
```

Reply:
it's a situation similar to the insights_id